File: backend/src/main/java/com/myfs/backend/model/SchoolClass.java
package com.myfs.backend.model;

import jakarta.persistence.*;

/**
 * Entity ánh xạ bảng school_class – lớp học.
 * homeroomTeacherId trỏ tới app_user của giáo viên chủ nhiệm.
 */
@Entity
@Table(name = "school_class")
public class SchoolClass {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "name", nullable = false, length = 50)
    private String name;

    @Column(name = "academic_year", nullable = false, length = 20)
    private String academicYear;

    @Column(name = "homeroom_teacher_id")
    private Integer homeroomTeacherId;

    public SchoolClass() {}

    public Integer getId()                { return id; }
    public String  getName()              { return name; }
    public String  getAcademicYear()      { return academicYear; }
    public Integer getHomeroomTeacherId() { return homeroomTeacherId; }

    public void setId(Integer v)                { this.id = v; }
    public void setName(String v)               { this.name = v; }
    public void setAcademicYear(String v)       { this.academicYear = v; }
    public void setHomeroomTeacherId(Integer v) { this.homeroomTeacherId = v; }
}

File: myfs/lib/models/school_class.dart
/// Lớp học – khớp entity SchoolClass bên backend.
class SchoolClass {
  final int id;
  final String name;
  final String academicYear;
  final int? homeroomTeacherId;

  const SchoolClass({
    required this.id,
    required this.name,
    required this.academicYear,
    this.homeroomTeacherId,
  });

  factory SchoolClass.fromJson(Map<String, dynamic> json) => SchoolClass(
    id: json['id'] as int,
    name: json['name'] as String,
    academicYear: json['academicYear'] as String,
    homeroomTeacherId: json['homeroomTeacherId'] as int?,
  );
}
